core/src/components/modal/modal.tsx

```diff
@@ -689,6 +689,21 @@ export class Modal implements ComponentInterface, OverlayInterface {
     return this.currentBreakpoint;
   }
 
+  private onHandleClick = () => {
+    const { breakpoints, currentBreakpoint } = this;
```

Comment:
Tabbing to the handle and pressing Enter multiple times causes the animation to jump.

https://user-images.githubusercontent.com/2721089/161641895-824bebbe-9e22-43fa-821c-6743f0d26c36.mov

Reply:
Adjusted this behavior slightly, so that advancing is tied to keyup instead of keydown. Setting the current breakpoint does not currently wait for the previous transition to finish. If you perform an action while a transition is inflight, it will disrupt the transition.

Reply:
Looks like it's still jumping if you mash Enter; I feel like if there's a transition in progress when you click/activate the handle, it should just do nothing.

Reply:
@amandaesmith3 to confirm, if the user performs any action (clicking the handle or pressing enter), while the sheet modal is actively transitioning, the expected behavior would be to perform no operation (skip the interaction)?

Reply:
Looking back at the code, `present` and `dismiss` will both wait for the current transition and then go through with the attempted interaction, so we should probably go with that instead here. Thinking on it, if I hit Enter twice quickly, having it only move to the next breakpoint once would feel off. 🤔

Reply:
Waiting for the transition feels like it would conflict with the design document for the API of moving to a breakpoint:
> If the user is in the middle of a gesture when this API is called, the user will be transitioned to the breakpoint value called by the API.

Unless we can cancel the current transition and immediately kick-off the new transition without a glitchy experience. I can do some more discovery on this.

Reply:
Updated the behavior, curious to both of your thoughts/feedback/alternatives. It now won't skip/jitter when you hold the enter key, but it will only performs moving to the next breakpoint when the current animation finishes. Hard to demonstrate in a recording.

Reply:
So when I try it, pressing Enter twice rapidly makes it only move once. It seems like it'll just eat your input if the animation is currently running. My guess is that since the current breakpoint doesn't actually update until after the animation finishes (triggered [here](https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/modal/gestures/sheet.ts#L215)), `nextBreakpoint` is calculated as the same thing for both Enter presses. Which would make the second press not really do anything.

That said, I'm okay with the behavior as-is. Up to you if you want to change it, though if we stick with it as-is, the code should probably be cleaned up to make it more clear.